### ISSUE TITLE
Build race: Error: not_found no_db_file: -/user/org.couchdb.user:tjholowaychuk

### ISSUE THREAD

**Issue by tj** [terinjokes/docker-npmjs#12]
oddly this didn't happen in my vm but I'm chucking npm up on ec2 and now it's barfing:

```
2013-12-01T19:53:38.712Z INFO,REQUEST 96.50.97.111 PUT /-/user/org.couchdb.user:tjholowaychuk 404
[Sun, 01 Dec 2013 19:54:18 GMT] [error] [<0.264.0>] Could not open file /usr/local/var/lib/couchdb/registry.couch: no such file or directory
[Sun, 01 Dec 2013 19:54:18 GMT] [info] [<0.219.0>] 172.17.0.22 - - PUT /registry/_design/app/_rewrite/-/user/org.couchdb.user:tjholowaychuk 404
```

**Comment by terinjokes** [comment]
`docker ps` to get the ID, then `docker logs $ID` to get the logs for that container. Doesn't look like something ran correctly.

Also, which version of docker do you have in EC2?

**Comment by terinjokes** [comment]
And any AMI information: I'll like to try to reproduce this if possible (since one of the points of Docker is to be reproducible)

**Comment by tj** [comment]
haha ya that's why it surprised me a bit, v0.7.0 on the vm and ec2, looks like couch initially freaks out, later breaking on some spam requests haha

```
[Sun, 01 Dec 2013 19:28:08 GMT] [info] [<0.32.0>] Apache CouchDB has started on http://0.0.0.0:5984/
[Sun, 01 Dec 2013 19:28:08 GMT] [info] [<0.32.0>] Apache CouchDB has started on https://0.0.0.0:6984/
[Sun, 01 Dec 2013 19:52:52 GMT] [info] [<0.32.0>] Apache CouchDB has started on http://0.0.0.0:5984/
[Sun, 01 Dec 2013 19:52:52 GMT] [info] [<0.32.0>] Apache CouchDB has started on https://0.0.0.0:6984/
[Sun, 01 Dec 2013 19:52:56 GMT] [error] [<0.168.0>] Could not open file /usr/local/var/lib/couchdb/registry.couch: no such file or directory
[Sun, 01 Dec 2013 19:52:56 GMT] [info] [<0.123.0>] 172.17.0.22 - - GET /registry/_design/app/_rewrite/ 404
2013-12-01T19:52:58.237Z INFO,REQUEST 172.17.42.1 GET / 200
[Sun, 01 Dec 2013 19:53:22 GMT] [error] [<0.200.0>] Could not open file /usr/local/var/lib/couchdb/registry.couch: no such file or directory
[Sun, 01 Dec 2013 19:53:22 GMT] [info] [<0.167.0>] 172.17.0.22 - - PUT /registry/_design/app/_rewrite/foo 404
2013-12-01T19:53:22.903Z INFO,REQUEST 96.50.97.111 PUT /foo 404
[Sun, 01 Dec 2013 19:53:38 GMT] [error] [<0.220.0>] Could not open file /usr/local/var/lib/couchdb/registry.couch: no such file or directory
[Sun, 01 Dec 2013 19:53:38 GMT] [info] [<0.199.0>] 172.17.0.22 - - PUT /registry/_design/app/_rewrite/-/user/org.couchdb.user:tjholowaychuk 404
2013-12-01T19:53:38.712Z INFO,REQUEST 96.50.97.111 PUT /-/user/org.couchdb.user:tjholowaychuk 404
[Sun, 01 Dec 2013 19:54:18 GMT] [error] [<0.264.0>] Could not open file /usr/local/var/lib/couchdb/registry.couch: no such file or directory
[Sun, 01 Dec 2013 19:54:18 GMT] [info] [<0.219.0>] 172.17.0.22 - - PUT /registry/_design/app/_rewrite/-/user/org.couchdb.user:tjholowaychuk 404
2013-12-01T19:54:18.687Z INFO,REQUEST 96.50.97.111 PUT /-/user/org.couchdb.user:tjholowaychuk 404
[Sun, 01 Dec 2013 21:04:40 GMT] [error] [<0.4492.0>] Could not open file /usr/local/var/lib/couchdb/registry.couch: no such file or directory
[Sun, 01 Dec 2013 21:04:40 GMT] [info] [<0.118.0>] 172.17.0.22 - - POST /registry/_design/app/_rewrite/cgi-bin/php/%63%67%69%6E/%70%68%70 404
...
```

**Comment by terinjokes** [comment]
Cool, I'll setup an EC2 when I get into work and see whats going on.

**Comment by tj** [comment]
ill do some digging too I'm not familiar with couch but seems like it should be some simple permissions issue or something

**Comment by tj** [comment]
interesting, my vm definitely has it just fine:

```
$ ls /usr/local/var/lib/couchdb/
_replicator.couch  _users.couch  registry.couch
```

and ec2:

```
$ ls /usr/local/var/lib/couchdb/
_replicator.couch  _users.couch
```

I'll try rebuilding the container

**Comment by terinjokes** [comment]
Yea. I don't think the docker container setup correctly. registry.couch is one of the very first things setup.

**Comment by tj** [comment]
ah damn yeah I should have watched that closer last time, it's definitely failing on build:

```
Step 1 : FROM sbisbee/couchdb:1.4
 ---> 127ea02eb1ad
Step 2 : MAINTAINER Terin Stock <terinjokes@gmail.com>
 ---> Running in 0f37c32da149
 ---> 67f62cda536a
Step 3 : ENV PATH /opt/node/bin/:$PATH
 ---> Running in 1971215ca94b
 ---> a848afefa1a4
Step 4 : RUN apt-get install -y curl git
 ---> Running in 63a391eb319d
Reading package lists...
Building dependency tree...
Reading state information...
The following extra packages will be installed:
  git-man less libbsd0 libclass-isa-perl libcurl3 libedit2 liberror-perl
  libexpat1 libgdbm3 libswitch-perl libx11-6 libx11-data libxau6 libxcb1
  libxdmcp6 libxext6 libxmuu1 netbase openssh-client patch perl perl-modules
  rsync xauth
Suggested packages:
  git-daemon-run git-daemon-sysvinit git-doc git-el git-arch git-cvs git-svn
  git-email git-gui gitk gitweb ssh-askpass libpam-ssh keychain monkeysphere
  openssh-blacklist openssh-blacklist-extra ed diffutils-doc perl-doc
  libterm-readline-gnu-perl libterm-readline-perl-perl libpod-plainer-perl
  openssh-server
Recommended packages:
  ssh-client
The following NEW packages will be installed:
  curl git git-man less libbsd0 libclass-isa-perl libcurl3 libedit2
  liberror-perl libexpat1 libgdbm3 libswitch-perl libx11-6 libx11-data libxau6
  libxcb1 libxdmcp6 libxext6 libxmuu1 netbase openssh-client patch perl
  perl-modules rsync xauth
0 upgraded, 26 newly installed, 0 to remove and 0 not upgraded.
Need to get 17.7 MB of archives.
After this operation, 56.3 MB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu/ precise/main libexpat1 amd64 2.0.1-7.2ubuntu1 [131 kB]
Get:2 http://archive.ubuntu.com/ubuntu/ precise/main libbsd0 amd64 0.3.0-2 [31.6 kB]
Get:3 http://archive.ubuntu.com/ubuntu/ precise/main libedit2 amd64 2.11-20080614-3ubuntu2 [70.3 kB]
Get:4 http://archive.ubuntu.com/ubuntu/ precise/main libgdbm3 amd64 1.8.3-10 [35.3 kB]
Get:5 http://archive.ubuntu.com/ubuntu/ precise/main libxau6 amd64 1:1.0.6-4 [8392 B]
Get:6 http://archive.ubuntu.com/ubuntu/ precise/main libxdmcp6 amd64 1:1.1.0-4 [12.7 kB]
Get:7 http://archive.ubuntu.com/ubuntu/ precise/main libxcb1 amd64 1.8.1-1 [44.7 kB]
Get:8 http://archive.ubuntu.com/ubuntu/ precise/main libx11-data all 2:1.4.99.1-0ubuntu2 [168 kB]
Get:9 http://archive.ubuntu.com/ubuntu/ precise/main libx11-6 amd64 2:1.4.99.1-0ubuntu2 [762 kB]
Get:10 http://archive.ubuntu.com/ubuntu/ precise/main libxext6 amd64 2:1.3.0-3build1 [33.2 kB]
Get:11 http://archive.ubuntu.com/ubuntu/ precise/main libxmuu1 amd64 2:1.1.0-3 [11.0 kB]
Get:12 http://archive.ubuntu.com/ubuntu/ precise/main libcurl3 amd64 7.22.0-3ubuntu4 [237 kB]
Get:13 http://archive.ubuntu.com/ubuntu/ precise/main less amd64 444-1ubuntu1 [114 kB]
Get:14 http://archive.ubuntu.com/ubuntu/ precise/main netbase all 4.47ubuntu1 [15.0 kB]
Get:15 http://archive.ubuntu.com/ubuntu/ precise/main openssh-client amd64 1:5.9p1-5ubuntu1 [943 kB]
Get:16 http://archive.ubuntu.com/ubuntu/ precise/main rsync amd64 3.0.9-1ubuntu1 [310 kB]
Get:17 http://archive.ubuntu.com/ubuntu/ precise/main xauth amd64 1:1.0.6-1 [26.2 kB]
Get:18 http://archive.ubuntu.com/ubuntu/ precise/main curl amd64 7.22.0-3ubuntu4 [138 kB]
Get:19 http://archive.ubuntu.com/ubuntu/ precise/main perl amd64 5.14.2-6ubuntu2 [4417 kB]
Get:20 http://archive.ubuntu.com/ubuntu/ precise/main libswitch-perl all 2.16-2 [19.2 kB]
Get:21 http://archive.ubuntu.com/ubuntu/ precise/main libclass-isa-perl all 0.36-3 [11.9 kB]
Get:22 http://archive.ubuntu.com/ubuntu/ precise/main perl-modules all 5.14.2-6ubuntu2 [3369 kB]
Get:23 http://archive.ubuntu.com/ubuntu/ precise/main liberror-perl all 0.17-1 [23.8 kB]
Get:24 http://archive.ubuntu.com/ubuntu/ precise/main git-man all 1:1.7.9.5-1 [630 kB]
Get:25 http://archive.ubuntu.com/ubuntu/ precise/main git amd64 1:1.7.9.5-1 [6087 kB]
Get:26 http://archive.ubuntu.com/ubuntu/ precise/main patch amd64 2.6.1-3 [80.2 kB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 17.7 MB in 9s (1901 kB/s)
Selecting previously unselected package libexpat1.
(Reading database ... 17125 files and directories currently installed.)
Unpacking libexpat1 (from .../libexpat1_2.0.1-7.2ubuntu1_amd64.deb) ...
Selecting previously unselected package libbsd0.
Unpacking libbsd0 (from .../libbsd0_0.3.0-2_amd64.deb) ...
Selecting previously unselected package libedit2.
Unpacking libedit2 (from .../libedit2_2.11-20080614-3ubuntu2_amd64.deb) ...
Selecting previously unselected package libgdbm3.
Unpacking libgdbm3 (from .../libgdbm3_1.8.3-10_amd64.deb) ...
Selecting previously unselected package libxau6.
Unpacking libxau6 (from .../libxau6_1%3a1.0.6-4_amd64.deb) ...
Selecting previously unselected package libxdmcp6.
Unpacking libxdmcp6 (from .../libxdmcp6_1%3a1.1.0-4_amd64.deb) ...
Selecting previously unselected package libxcb1.
Unpacking libxcb1 (from .../libxcb1_1.8.1-1_amd64.deb) ...
Selecting previously unselected package libx11-data.
Unpacking libx11-data (from .../libx11-data_2%3a1.4.99.1-0ubuntu2_all.deb) ...
Selecting previously unselected package libx11-6.
Unpacking libx11-6 (from .../libx11-6_2%3a1.4.99.1-0ubuntu2_amd64.deb) ...
Selecting previously unselected package libxext6.
Unpacking libxext6 (from .../libxext6_2%3a1.3.0-3build1_amd64.deb) ...
Selecting previously unselected package libxmuu1.
Unpacking libxmuu1 (from .../libxmuu1_2%3a1.1.0-3_amd64.deb) ...
Selecting previously unselected package libcurl3.
Unpacking libcurl3 (from .../libcurl3_7.22.0-3ubuntu4_amd64.deb) ...
Selecting previously unselected package less.
Unpacking less (from .../less_444-1ubuntu1_amd64.deb) ...
Selecting previously unselected package netbase.
Unpacking netbase (from .../netbase_4.47ubuntu1_all.deb) ...
Selecting previously unselected package openssh-client.
Unpacking openssh-client (from .../openssh-client_1%3a5.9p1-5ubuntu1_amd64.deb) ...
Selecting previously unselected package rsync.
Unpacking rsync (from .../rsync_3.0.9-1ubuntu1_amd64.deb) ...
Selecting previously unselected package xauth.
Unpacking xauth (from .../xauth_1%3a1.0.6-1_amd64.deb) ...
Selecting previously unselected package curl.
Unpacking curl (from .../curl_7.22.0-3ubuntu4_amd64.deb) ...
Selecting previously unselected package perl.
Unpacking perl (from .../perl_5.14.2-6ubuntu2_amd64.deb) ...
Selecting previously unselected package libswitch-perl.
Unpacking libswitch-perl (from .../libswitch-perl_2.16-2_all.deb) ...
Selecting previously unselected package libclass-isa-perl.
Unpacking libclass-isa-perl (from .../libclass-isa-perl_0.36-3_all.deb) ...
Selecting previously unselected package perl-modules.
Unpacking perl-modules (from .../perl-modules_5.14.2-6ubuntu2_all.deb) ...
Selecting previously unselected package liberror-perl.
Unpacking liberror-perl (from .../liberror-perl_0.17-1_all.deb) ...
Selecting previously unselected package git-man.
Unpacking git-man (from .../git-man_1%3a1.7.9.5-1_all.deb) ...
Selecting previously unselected package git.
Unpacking git (from .../git_1%3a1.7.9.5-1_amd64.deb) ...
Selecting previously unselected package patch.
Unpacking patch (from .../patch_2.6.1-3_amd64.deb) ...
Setting up libexpat1 (2.0.1-7.2ubuntu1) ...
Setting up libbsd0 (0.3.0-2) ...
Setting up libedit2 (2.11-20080614-3ubuntu2) ...
Setting up libgdbm3 (1.8.3-10) ...
Setting up libxau6 (1:1.0.6-4) ...
Setting up libxdmcp6 (1:1.1.0-4) ...
Setting up libxcb1 (1.8.1-1) ...
Setting up libx11-data (2:1.4.99.1-0ubuntu2) ...
Setting up libx11-6 (2:1.4.99.1-0ubuntu2) ...
Setting up libxext6 (2:1.3.0-3build1) ...
Setting up libxmuu1 (2:1.1.0-3) ...
Setting up libcurl3 (7.22.0-3ubuntu4) ...
Setting up less (444-1ubuntu1) ...
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (This frontend requires a controlling tty.)
debconf: falling back to frontend: Teletype
Setting up netbase (4.47ubuntu1) ...
Setting up openssh-client (1:5.9p1-5ubuntu1) ...
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (This frontend requires a controlling tty.)
debconf: falling back to frontend: Teletype
Setting up rsync (3.0.9-1ubuntu1) ...
 Removing any system startup links for /etc/init.d/rsync ...
Setting up xauth (1:1.0.6-1) ...
Setting up curl (7.22.0-3ubuntu4) ...
Setting up libclass-isa-perl (0.36-3) ...
Setting up git-man (1:1.7.9.5-1) ...
Setting up patch (2.6.1-3) ...
Setting up perl-modules (5.14.2-6ubuntu2) ...
Setting up perl (5.14.2-6ubuntu2) ...
update-alternatives: using /usr/bin/prename to provide /usr/bin/rename (rename) in auto mode.
Setting up libswitch-perl (2.16-2) ...
Setting up liberror-perl (0.17-1) ...
Setting up git (1:1.7.9.5-1) ...
Processing triggers for libc-bin ...
ldconfig deferred processing now taking place
 ---> 347d2af0e7a4
Step 5 : RUN mkdir -p /opt/node
 ---> Running in 0f9b13fbc2e2
 ---> 02aa125e9cec
Step 6 : RUN curl -L# http://nodejs.org/dist/v0.10.21/node-v0.10.21-linux-x64.tar.gz|tar -zx --strip 1 -C /opt/node
 ---> Running in 8dda4ae3a69b
######################################################################## 100.0%
 ---> 91c81f1e364a
Step 7 : RUN git clone https://github.com/isaacs/npmjs.org /opt/npmjs
 ---> Running in 87591dec8c0d
Cloning into '/opt/npmjs'...
 ---> fe0e2c989346
Step 8 : RUN cd /opt/npmjs; git checkout ea8e7a533ea595db79b24f12c76b62c3889b43e8
 ---> Running in e0a28bccd140
Note: checking out 'ea8e7a533ea595db79b24f12c76b62c3889b43e8'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b new_branch_name

HEAD is now at ea8e7a5... you may not delete npm
 ---> a0278f7ae1a8
Step 9 : RUN npm install couchapp@0.10.x -g
 ---> Running in 27ea79c1f5f4
npm http GET https://registry.npmjs.org/couchapp
npm http 200 https://registry.npmjs.org/couchapp
npm http GET https://registry.npmjs.org/couchapp/-/couchapp-0.10.0.tgz
npm http 200 https://registry.npmjs.org/couchapp/-/couchapp-0.10.0.tgz
npm http GET https://registry.npmjs.org/connect
npm http GET https://registry.npmjs.org/http-proxy
npm http GET https://registry.npmjs.org/nano
npm http GET https://registry.npmjs.org/url
npm http GET https://registry.npmjs.org/coffee-script
npm http GET https://registry.npmjs.org/watch
npm http GET https://registry.npmjs.org/request
npm http 200 https://registry.npmjs.org/url
npm http GET https://registry.npmjs.org/url/-/url-0.7.9.tgz
npm http 200 https://registry.npmjs.org/coffee-script
npm http GET https://registry.npmjs.org/coffee-script/-/coffee-script-1.6.3.tgz
npm http 200 https://registry.npmjs.org/watch
npm http GET https://registry.npmjs.org/watch/-/watch-0.8.0.tgz
npm http 200 https://registry.npmjs.org/connect
npm http GET https://registry.npmjs.org/connect/-/connect-2.11.2.tgz
npm http 200 https://registry.npmjs.org/url/-/url-0.7.9.tgz
npm http 200 https://registry.npmjs.org/http-proxy
npm http GET https://registry.npmjs.org/http-proxy/-/http-proxy-0.10.3.tgz
npm http 200 https://registry.npmjs.org/coffee-script/-/coffee-script-1.6.3.tgz
npm http 200 https://registry.npmjs.org/watch/-/watch-0.8.0.tgz
npm http 200 https://registry.npmjs.org/request
npm http GET https://registry.npmjs.org/request/-/request-2.27.0.tgz
npm http 200 https://registry.npmjs.org/connect/-/connect-2.11.2.tgz
npm http 200 https://registry.npmjs.org/http-proxy/-/http-proxy-0.10.3.tgz
npm http 200 https://registry.npmjs.org/nano
npm http GET https://registry.npmjs.org/nano/-/nano-4.2.1.tgz
npm http 200 https://registry.npmjs.org/request/-/request-2.27.0.tgz
npm http 200 https://registry.npmjs.org/nano/-/nano-4.2.1.tgz
npm http GET https://registry.npmjs.org/querystring
npm http GET https://registry.npmjs.org/punycode
npm http GET https://registry.npmjs.org/qs
npm http GET https://registry.npmjs.org/json-stringify-safe
npm http GET https://registry.npmjs.org/tunnel-agent
npm http GET https://registry.npmjs.org/forever-agent
npm http GET https://registry.npmjs.org/http-signature
npm http GET https://registry.npmjs.org/hawk
npm http GET https://registry.npmjs.org/aws-sign
npm http GET https://registry.npmjs.org/oauth-sign
npm http GET https://registry.npmjs.org/cookie-jar
npm http GET https://registry.npmjs.org/node-uuid
npm http GET https://registry.npmjs.org/mime
npm http GET https://registry.npmjs.org/form-data
npm http 200 https://registry.npmjs.org/querystring
npm http GET https://registry.npmjs.org/pkginfo
npm http GET https://registry.npmjs.org/utile
npm http GET https://registry.npmjs.org/colors
npm http GET https://registry.npmjs.org/optimist
npm http GET https://registry.npmjs.org/querystring/-/querystring-0.1.0.tgz
npm http GET https://registry.npmjs.org/follow
npm http GET https://registry.npmjs.org/errs
npm http GET https://registry.npmjs.org/underscore
npm http 200 https://registry.npmjs.org/punycode
npm http GET https://registry.npmjs.org/punycode/-/punycode-1.0.0.tgz
npm http GET https://registry.npmjs.org/qs/0.6.5
npm http GET https://registry.npmjs.org/cookie-signature/1.0.1
npm http GET https://registry.npmjs.org/buffer-crc32/0.2.1
npm http GET https://registry.npmjs.org/cookie/0.1.0
npm http GET https://registry.npmjs.org/send/0.1.4
npm http GET https://registry.npmjs.org/bytes/0.2.1
npm http GET https://registry.npmjs.org/fresh/0.2.0
npm http GET https://registry.npmjs.org/pause/0.0.1
npm http GET https://registry.npmjs.org/uid2/0.0.3
npm http GET https://registry.npmjs.org/debug
npm http GET https://registry.npmjs.org/methods/0.1.0
npm http GET https://registry.npmjs.org/raw-body/1.1.2
npm http GET https://registry.npmjs.org/negotiator/0.3.0
npm http GET https://registry.npmjs.org/multiparty/2.2.0
npm http 200 https://registry.npmjs.org/forever-agent
npm http GET https://registry.npmjs.org/forever-agent/-/forever-agent-0.5.0.tgz
npm http 200 https://registry.npmjs.org/http-signature
npm http GET https://registry.npmjs.org/http-signature/-/http-signature-0.10.0.tgz
npm http 200 https://registry.npmjs.org/aws-sign
npm http GET https://registry.npmjs.org/aws-sign/-/aws-sign-0.3.0.tgz
npm http 200 https://registry.npmjs.org/json-stringify-safe
npm http GET https://registry.npmjs.org/json-stringify-safe/-/json-stringify-safe-5.0.0.tgz
npm http 200 https://registry.npmjs.org/tunnel-agent
npm http GET https://registry.npmjs.org/tunnel-agent/-/tunnel-agent-0.3.0.tgz
npm http 200 https://registry.npmjs.org/querystring/-/querystring-0.1.0.tgz
npm http 200 https://registry.npmjs.org/punycode/-/punycode-1.0.0.tgz
npm http 200 https://registry.npmjs.org/oauth-sign
npm http 200 https://registry.npmjs.org/qs
npm http GET https://registry.npmjs.org/oauth-sign/-/oauth-sign-0.3.0.tgz
npm http 200 https://registry.npmjs.org/cookie-jar
npm http GET https://registry.npmjs.org/cookie-jar/-/cookie-jar-0.3.0.tgz
npm http 200 https://registry.npmjs.org/json-stringify-safe/-/json-stringify-safe-5.0.0.tgz
npm http 200 https://registry.npmjs.org/forever-agent/-/forever-agent-0.5.0.tgz
npm http 200 https://registry.npmjs.org/node-uuid
npm http GET https://registry.npmjs.org/node-uuid/-/node-uuid-1.4.1.tgz
npm http 200 https://registry.npmjs.org/mime
npm http 200 https://registry.npmjs.org/http-signature/-/http-signature-0.10.0.tgz
npm http GET https://registry.npmjs.org/mime/-/mime-1.2.11.tgz
npm http 200 https://registry.npmjs.org/form-data
npm http GET https://registry.npmjs.org/form-data/-/form-data-0.1.2.tgz
npm http 200 https://registry.npmjs.org/pkginfo
npm http GET https://registry.npmjs.org/pkginfo/-/pkginfo-0.2.3.tgz
npm http 200 https://registry.npmjs.org/tunnel-agent/-/tunnel-agent-0.3.0.tgz
npm http 200 https://registry.npmjs.org/oauth-sign/-/oauth-sign-0.3.0.tgz
npm http 200 https://registry.npmjs.org/cookie-jar/-/cookie-jar-0.3.0.tgz
npm http 200 https://registry.npmjs.org/colors
npm http 200 https://registry.npmjs.org/aws-sign/-/aws-sign-0.3.0.tgz
npm http GET https://registry.npmjs.org/colors/-/colors-0.6.2.tgz
npm http 200 https://registry.npmjs.org/utile
npm http GET https://registry.npmjs.org/utile/-/utile-0.1.7.tgz
npm http 200 https://registry.npmjs.org/node-uuid/-/node-uuid-1.4.1.tgz
npm http 200 https://registry.npmjs.org/optimist
npm http 200 https://registry.npmjs.org/mime/-/mime-1.2.11.tgz
npm http 200 https://registry.npmjs.org/form-data/-/form-data-0.1.2.tgz
npm http GET https://registry.npmjs.org/optimist/-/optimist-0.3.7.tgz
npm http 200 https://registry.npmjs.org/pkginfo/-/pkginfo-0.2.3.tgz
npm http 200 https://registry.npmjs.org/errs
npm http 200 https://registry.npmjs.org/underscore
npm http GET https://registry.npmjs.org/errs/-/errs-0.2.4.tgz
npm http 200 https://registry.npmjs.org/colors/-/colors-0.6.2.tgz
npm http 200 https://registry.npmjs.org/follow
npm http 200 https://registry.npmjs.org/qs/0.6.5
npm http 200 https://registry.npmjs.org/utile/-/utile-0.1.7.tgz
npm http 200 https://registry.npmjs.org/optimist/-/optimist-0.3.7.tgz
npm http GET https://registry.npmjs.org/underscore/-/underscore-1.5.2.tgz
npm http GET https://registry.npmjs.org/qs/-/qs-0.6.5.tgz
npm http 200 https://registry.npmjs.org/cookie-signature/1.0.1
npm http 200 https://registry.npmjs.org/errs/-/errs-0.2.4.tgz
npm http GET https://registry.npmjs.org/follow/-/follow-0.10.1.tgz
npm http 200 https://registry.npmjs.org/buffer-crc32/0.2.1
npm http GET https://registry.npmjs.org/cookie-signature/-/cookie-signature-1.0.1.tgz
npm http 200 https://registry.npmjs.org/cookie/0.1.0
npm http GET https://registry.npmjs.org/buffer-crc32/-/buffer-crc32-0.2.1.tgz
npm http 200 https://registry.npmjs.org/qs/-/qs-0.6.5.tgz
npm http 200 https://registry.npmjs.org/send/0.1.4
npm http GET https://registry.npmjs.org/cookie/-/cookie-0.1.0.tgz
npm http GET https://registry.npmjs.org/send/-/send-0.1.4.tgz
npm http 200 https://registry.npmjs.org/pause/0.0.1
npm http 200 https://registry.npmjs.org/bytes/0.2.1
npm http 200 https://registry.npmjs.org/follow/-/follow-0.10.1.tgz
npm http GET https://registry.npmjs.org/pause/-/pause-0.0.1.tgz
npm http 200 https://registry.npmjs.org/uid2/0.0.3
npm http 200 https://registry.npmjs.org/fresh/0.2.0
npm http GET https://registry.npmjs.org/bytes/-/bytes-0.2.1.tgz
npm http GET https://registry.npmjs.org/uid2/-/uid2-0.0.3.tgz
npm http GET https://registry.npmjs.org/fresh/-/fresh-0.2.0.tgz
npm http 200 https://registry.npmjs.org/cookie-signature/-/cookie-signature-1.0.1.tgz
npm http 200 https://registry.npmjs.org/buffer-crc32/-/buffer-crc32-0.2.1.tgz
npm http 200 https://registry.npmjs.org/methods/0.1.0
npm http 200 https://registry.npmjs.org/debug
npm http 200 https://registry.npmjs.org/hawk
npm http GET https://registry.npmjs.org/methods/-/methods-0.1.0.tgz
npm http 200 https://registry.npmjs.org/raw-body/1.1.2
npm http 200 https://registry.npmjs.org/negotiator/0.3.0
npm http GET https://registry.npmjs.org/debug/-/debug-0.7.4.tgz
npm http GET https://registry.npmjs.org/hawk/-/hawk-1.0.0.tgz
npm http 200 https://registry.npmjs.org/cookie/-/cookie-0.1.0.tgz
npm http 200 https://registry.npmjs.org/send/-/send-0.1.4.tgz
npm http GET https://registry.npmjs.org/raw-body/-/raw-body-1.1.2.tgz
npm http 200 https://registry.npmjs.org/multiparty/2.2.0
npm http GET https://registry.npmjs.org/negotiator/-/negotiator-0.3.0.tgz
npm http GET https://registry.npmjs.org/multiparty/-/multiparty-2.2.0.tgz
npm http 200 https://registry.npmjs.org/underscore/-/underscore-1.5.2.tgz
npm http 200 https://registry.npmjs.org/pause/-/pause-0.0.1.tgz
npm http 200 https://registry.npmjs.org/bytes/-/bytes-0.2.1.tgz
npm http 200 https://registry.npmjs.org/uid2/-/uid2-0.0.3.tgz
npm http 200 https://registry.npmjs.org/fresh/-/fresh-0.2.0.tgz
npm http 200 https://registry.npmjs.org/debug/-/debug-0.7.4.tgz
npm http 200 https://registry.npmjs.org/methods/-/methods-0.1.0.tgz
npm http 200 https://registry.npmjs.org/hawk/-/hawk-1.0.0.tgz
npm http 200 https://registry.npmjs.org/raw-body/-/raw-body-1.1.2.tgz
npm http 200 https://registry.npmjs.org/negotiator/-/negotiator-0.3.0.tgz
npm http 200 https://registry.npmjs.org/multiparty/-/multiparty-2.2.0.tgz
npm http GET https://registry.npmjs.org/async
npm http GET https://registry.npmjs.org/deep-equal
npm http GET https://registry.npmjs.org/i
npm http GET https://registry.npmjs.org/mkdirp
npm http GET https://registry.npmjs.org/ncp
npm http GET https://registry.npmjs.org/rimraf
npm http GET https://registry.npmjs.org/wordwrap
npm http 200 https://registry.npmjs.org/deep-equal
npm http GET https://registry.npmjs.org/deep-equal/-/deep-equal-0.1.0.tgz
npm http 200 https://registry.npmjs.org/mkdirp
npm http 200 https://registry.npmjs.org/i
npm http GET https://registry.npmjs.org/mkdirp/-/mkdirp-0.3.5.tgz
npm http GET https://registry.npmjs.org/i/-/i-0.3.2.tgz
npm http 200 https://registry.npmjs.org/ncp
npm http GET https://registry.npmjs.org/ncp/-/ncp-0.2.7.tgz
npm http 200 https://registry.npmjs.org/wordwrap
npm http GET https://registry.npmjs.org/wordwrap/-/wordwrap-0.0.2.tgz
npm http 200 https://registry.npmjs.org/rimraf
npm http GET https://registry.npmjs.org/rimraf/-/rimraf-1.0.9.tgz
npm http GET https://registry.npmjs.org/browser-request
npm http GET https://registry.npmjs.org/request
npm http GET https://registry.npmjs.org/combined-stream
npm http GET https://registry.npmjs.org/async
npm http 200 https://registry.npmjs.org/deep-equal/-/deep-equal-0.1.0.tgz
npm http GET https://registry.npmjs.org/assert-plus/0.1.2
npm http GET https://registry.npmjs.org/asn1/0.1.11
npm http GET https://registry.npmjs.org/ctype/0.5.2
npm http 200 https://registry.npmjs.org/mkdirp/-/mkdirp-0.3.5.tgz
npm http 200 https://registry.npmjs.org/i/-/i-0.3.2.tgz
npm http 200 https://registry.npmjs.org/wordwrap/-/wordwrap-0.0.2.tgz
npm http 200 https://registry.npmjs.org/ncp/-/ncp-0.2.7.tgz
npm http 200 https://registry.npmjs.org/rimraf/-/rimraf-1.0.9.tgz
npm http 304 https://registry.npmjs.org/request
npm http GET https://registry.npmjs.org/request/-/request-2.16.6.tgz
npm http 200 https://registry.npmjs.org/browser-request
npm http GET https://registry.npmjs.org/browser-request/-/browser-request-0.3.1.tgz
npm http 200 https://registry.npmjs.org/combined-stream
npm http GET https://registry.npmjs.org/combined-stream/-/combined-stream-0.0.4.tgz
npm http 200 https://registry.npmjs.org/assert-plus/0.1.2
npm http GET https://registry.npmjs.org/assert-plus/-/assert-plus-0.1.2.tgz
npm http 200 https://registry.npmjs.org/asn1/0.1.11
npm http 200 https://registry.npmjs.org/ctype/0.5.2
npm http GET https://registry.npmjs.org/asn1/-/asn1-0.1.11.tgz
npm http GET https://registry.npmjs.org/ctype/-/ctype-0.5.2.tgz
npm http GET https://registry.npmjs.org/sntp
npm http GET https://registry.npmjs.org/cryptiles
npm http GET https://registry.npmjs.org/hoek
npm http GET https://registry.npmjs.org/boom
npm http 200 https://registry.npmjs.org/async
npm http GET https://registry.npmjs.org/async/-/async-0.1.22.tgz
npm http 200 https://registry.npmjs.org/request/-/request-2.16.6.tgz
npm http 200 https://registry.npmjs.org/browser-request/-/browser-request-0.3.1.tgz
npm http 200 https://registry.npmjs.org/boom
npm http GET https://registry.npmjs.org/boom/-/boom-0.4.2.tgz
npm http 200 https://registry.npmjs.org/combined-stream/-/combined-stream-0.0.4.tgz
npm http 200 https://registry.npmjs.org/assert-plus/-/assert-plus-0.1.2.tgz
npm http 200 https://registry.npmjs.org/asn1/-/asn1-0.1.11.tgz
npm http 200 https://registry.npmjs.org/ctype/-/ctype-0.5.2.tgz
npm http 200 https://registry.npmjs.org/async/-/async-0.1.22.tgz
npm http 200 https://registry.npmjs.org/boom/-/boom-0.4.2.tgz
npm http 200 https://registry.npmjs.org/cryptiles
npm http GET https://registry.npmjs.org/cryptiles/-/cryptiles-0.2.2.tgz
npm http 200 https://registry.npmjs.org/sntp
npm http GET https://registry.npmjs.org/sntp/-/sntp-0.2.4.tgz
npm http 200 https://registry.npmjs.org/cryptiles/-/cryptiles-0.2.2.tgz
npm http 200 https://registry.npmjs.org/async
npm http 200 https://registry.npmjs.org/sntp/-/sntp-0.2.4.tgz
npm http GET https://registry.npmjs.org/async/-/async-0.2.9.tgz
npm http 200 https://registry.npmjs.org/hoek
npm http GET https://registry.npmjs.org/hoek/-/hoek-0.9.1.tgz
npm http 200 https://registry.npmjs.org/async/-/async-0.2.9.tgz
npm http GET https://registry.npmjs.org/delayed-stream/0.0.5
npm http 200 https://registry.npmjs.org/hoek/-/hoek-0.9.1.tgz
npm http GET https://registry.npmjs.org/hawk
npm http GET https://registry.npmjs.org/node-uuid
npm http GET https://registry.npmjs.org/cookie-jar
npm http GET https://registry.npmjs.org/aws-sign
npm http GET https://registry.npmjs.org/oauth-sign
npm http GET https://registry.npmjs.org/forever-agent
npm http GET https://registry.npmjs.org/tunnel-agent
npm http GET https://registry.npmjs.org/json-stringify-safe
npm http GET https://registry.npmjs.org/qs
npm http GET https://registry.npmjs.org/form-data
npm http GET https://registry.npmjs.org/mime
npm http 200 https://registry.npmjs.org/delayed-stream/0.0.5
npm http GET https://registry.npmjs.org/delayed-stream/-/delayed-stream-0.0.5.tgz
npm http 304 https://registry.npmjs.org/oauth-sign
npm http GET https://registry.npmjs.org/oauth-sign/-/oauth-sign-0.2.0.tgz
npm http 304 https://registry.npmjs.org/forever-agent
npm http GET https://registry.npmjs.org/forever-agent/-/forever-agent-0.2.0.tgz
npm http 304 https://registry.npmjs.org/tunnel-agent
npm http GET https://registry.npmjs.org/tunnel-agent/-/tunnel-agent-0.2.0.tgz
npm http 304 https://registry.npmjs.org/json-stringify-safe
npm http GET https://registry.npmjs.org/json-stringify-safe/-/json-stringify-safe-3.0.0.tgz
npm http 304 https://registry.npmjs.org/qs
npm http GET https://registry.npmjs.org/qs/-/qs-0.5.6.tgz
npm http 304 https://registry.npmjs.org/cookie-jar
npm http 304 https://registry.npmjs.org/hawk
npm http 304 https://registry.npmjs.org/node-uuid
npm http 304 https://registry.npmjs.org/aws-sign
npm http 304 https://registry.npmjs.org/form-data
npm http GET https://registry.npmjs.org/cookie-jar/-/cookie-jar-0.2.0.tgz
npm http GET https://registry.npmjs.org/hawk/-/hawk-0.10.2.tgz
npm http GET https://registry.npmjs.org/aws-sign/-/aws-sign-0.2.0.tgz
npm http 304 https://registry.npmjs.org/mime
npm http 200 https://registry.npmjs.org/delayed-stream/-/delayed-stream-0.0.5.tgz
npm http GET https://registry.npmjs.org/form-data/-/form-data-0.0.10.tgz
npm http 200 https://registry.npmjs.org/qs/-/qs-0.5.6.tgz
npm http 200 https://registry.npmjs.org/forever-agent/-/forever-agent-0.2.0.tgz
npm http 200 https://registry.npmjs.org/tunnel-agent/-/tunnel-agent-0.2.0.tgz
npm http 200 https://registry.npmjs.org/json-stringify-safe/-/json-stringify-safe-3.0.0.tgz
npm http 200 https://registry.npmjs.org/cookie-jar/-/cookie-jar-0.2.0.tgz
npm http 200 https://registry.npmjs.org/hawk/-/hawk-0.10.2.tgz
npm http 200 https://registry.npmjs.org/aws-sign/-/aws-sign-0.2.0.tgz
npm http 200 https://registry.npmjs.org/form-data/-/form-data-0.0.10.tgz
npm http 200 https://registry.npmjs.org/oauth-sign/-/oauth-sign-0.2.0.tgz
npm WARN engine hawk@0.10.2: wanted: {"node":"0.8.x"} (current: {"node":"v0.10.21","npm":"1.3.11"})
npm http GET https://registry.npmjs.org/combined-stream
npm http GET https://registry.npmjs.org/async
npm http GET https://registry.npmjs.org/boom
npm http GET https://registry.npmjs.org/sntp
npm http GET https://registry.npmjs.org/cryptiles
npm http GET https://registry.npmjs.org/hoek/-/hoek-0.7.6.tgz
npm http 304 https://registry.npmjs.org/combined-stream
npm http 304 https://registry.npmjs.org/async
npm http 304 https://registry.npmjs.org/boom
npm http 304 https://registry.npmjs.org/cryptiles
npm http 304 https://registry.npmjs.org/sntp
npm http 200 https://registry.npmjs.org/hoek/-/hoek-0.7.6.tgz
npm http GET https://registry.npmjs.org/boom/-/boom-0.3.8.tgz
npm http GET https://registry.npmjs.org/cryptiles/-/cryptiles-0.1.3.tgz
npm http GET https://registry.npmjs.org/sntp/-/sntp-0.1.4.tgz
npm http 200 https://registry.npmjs.org/sntp/-/sntp-0.1.4.tgz
npm http 200 https://registry.npmjs.org/cryptiles/-/cryptiles-0.1.3.tgz
npm http 200 https://registry.npmjs.org/boom/-/boom-0.3.8.tgz
npm WARN engine hoek@0.7.6: wanted: {"node":"0.8.x"} (current: {"node":"v0.10.21","npm":"1.3.11"})
npm WARN engine cryptiles@0.1.3: wanted: {"node":"0.8.x"} (current: {"node":"v0.10.21","npm":"1.3.11"})
npm WARN engine sntp@0.1.4: wanted: {"node":"0.8.x"} (current: {"node":"v0.10.21","npm":"1.3.11"})
npm WARN engine boom@0.3.8: wanted: {"node":"0.8.x"} (current: {"node":"v0.10.21","npm":"1.3.11"})
npm http GET https://registry.npmjs.org/range-parser/0.0.4
npm http GET https://registry.npmjs.org/mime
npm http 200 https://registry.npmjs.org/range-parser/0.0.4
npm http 304 https://registry.npmjs.org/mime
npm http GET https://registry.npmjs.org/range-parser/-/range-parser-0.0.4.tgz
npm http 200 https://registry.npmjs.org/range-parser/-/range-parser-0.0.4.tgz
npm http GET https://registry.npmjs.org/readable-stream
npm http GET https://registry.npmjs.org/stream-counter
npm http 200 https://registry.npmjs.org/stream-counter
npm http GET https://registry.npmjs.org/stream-counter/-/stream-counter-0.2.0.tgz
npm http 200 https://registry.npmjs.org/stream-counter/-/stream-counter-0.2.0.tgz
npm http 200 https://registry.npmjs.org/readable-stream
npm http GET https://registry.npmjs.org/readable-stream/-/readable-stream-1.1.9.tgz
npm http 200 https://registry.npmjs.org/readable-stream/-/readable-stream-1.1.9.tgz
npm http GET https://registry.npmjs.org/core-util-is
npm http GET https://registry.npmjs.org/debuglog/0.0.2
npm http 200 https://registry.npmjs.org/core-util-is
npm http 200 https://registry.npmjs.org/debuglog/0.0.2
npm http GET https://registry.npmjs.org/core-util-is/-/core-util-is-1.0.0.tgz
npm http GET https://registry.npmjs.org/debuglog/-/debuglog-0.0.2.tgz
npm http 200 https://registry.npmjs.org/core-util-is/-/core-util-is-1.0.0.tgz
npm http 200 https://registry.npmjs.org/debuglog/-/debuglog-0.0.2.tgz
/opt/node/bin/couchapp -> /opt/node/lib/node_modules/couchapp/bin.js
couchapp@0.10.0 /opt/node/lib/node_modules/couchapp
├── watch@0.8.0
├── coffee-script@1.6.3
├── http-proxy@0.10.3 (colors@0.6.2, pkginfo@0.2.3, optimist@0.3.7, utile@0.1.7)
├── request@2.27.0 (json-stringify-safe@5.0.0, forever-agent@0.5.0, aws-sign@0.3.0, qs@0.6.5, tunnel-agent@0.3.0, oauth-sign@0.3.0, cookie-jar@0.3.0, node-uuid@1.4.1, mime@1.2.11, http-signature@0.10.0, hawk@1.0.0, form-data@0.1.2)
├── nano@4.2.1 (underscore@1.5.2, errs@0.2.4, follow@0.10.1)
├── connect@2.11.2 (uid2@0.0.3, methods@0.1.0, cookie-signature@1.0.1, pause@0.0.1, fresh@0.2.0, debug@0.7.4, qs@0.6.5, bytes@0.2.1, raw-body@1.1.2, buffer-crc32@0.2.1, cookie@0.1.0, negotiator@0.3.0, send@0.1.4, multiparty@2.2.0)
└── url@0.7.9 (querystring@0.1.0, punycode@1.0.0)
 ---> 60fe8fc5a513
Step 10 : RUN cd /opt/npmjs; npm link couchapp; npm install semver
 ---> Running in 2eaf6a9a64d8
/opt/npmjs/node_modules/couchapp -> /opt/node/lib/node_modules/couchapp
npm WARN package.json npmjs.org@1.0.1 No repository field.
npm WARN package.json couchapp@0.10.0 No repository field.
npm http GET https://registry.npmjs.org/semver
npm http 200 https://registry.npmjs.org/semver
npm http GET https://registry.npmjs.org/semver/-/semver-2.0.11.tgz
npm http 200 https://registry.npmjs.org/semver/-/semver-2.0.11.tgz
semver@2.0.11 node_modules/semver
 ---> 54e8b1a2a05c
Step 11 : RUN echo "[httpd]\nsecure_rewrites = false" >> /usr/local/etc/couchdb/local.d/secure_rewrites.ini
 ---> Running in b70e65e6f2e5
 ---> 1561a31c1d80
Step 12 : RUN cd /opt/npmjs; couchdb -b; sleep 1; curl -X PUT http://localhost:5984/registry; sleep 1; couchdb -d;
 ---> Running in f6acd68a37dd
Apache CouchDB has started, time to relax.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) couldn't connect to host
Apache CouchDB has been shutdown.
 ---> 00a4d588d797
Step 13 : RUN cd /opt/npmjs; couchdb -b; sleep 1; couchapp push registry/shadow.js http://localhost:5984/registry; sleep 1; couchapp push registry/app.js http://localhost:5984/registry; sleep 1; couchdb -d
 ---> Running in 1b2e4cff40e0
Apache CouchDB has started, time to relax.
Preparing.
Serializing.
PUT http://localhost:5984/registry/_design/ghost

Error: Could not push document
{"error":"not_found","reason":"no_db_file"}

    at Request._callback (/opt/node/lib/node_modules/couchapp/main.js:155:42)
    at Request.self.callback (/opt/node/lib/node_modules/couchapp/node_modules/request/request.js:129:22)
    at Request.EventEmitter.emit (events.js:98:17)
    at Request.<anonymous> (/opt/node/lib/node_modules/couchapp/node_modules/request/request.js:873:14)
    at Request.EventEmitter.emit (events.js:117:20)
    at IncomingMessage.<anonymous> (/opt/node/lib/node_modules/couchapp/node_modules/request/request.js:824:12)
    at IncomingMessage.EventEmitter.emit (events.js:117:20)
    at _stream_readable.js:920:16
    at process._tickCallback (node.js:415:13)
Preparing.
Serializing.
PUT http://localhost:5984/registry/_design/scratch

Error: Could not push document
{"error":"not_found","reason":"no_db_file"}

    at Request._callback (/opt/node/lib/node_modules/couchapp/main.js:155:42)
    at Request.self.callback (/opt/node/lib/node_modules/couchapp/node_modules/request/request.js:129:22)
    at Request.EventEmitter.emit (events.js:98:17)
    at Request.<anonymous> (/opt/node/lib/node_modules/couchapp/node_modules/request/request.js:873:14)
    at Request.EventEmitter.emit (events.js:117:20)
    at IncomingMessage.<anonymous> (/opt/node/lib/node_modules/couchapp/node_modules/request/request.js:824:12)
    at IncomingMessage.EventEmitter.emit (events.js:117:20)
    at _stream_readable.js:920:16
    at process._tickCallback (node.js:415:13)
Apache CouchDB has been shutdown.
 ---> 64747913d364
Step 14 : RUN cd /opt/npmjs; npm set _npmjs.org:couch=http://localhost:5984/registry
 ---> Running in 00b8e4b59fbb
 ---> 1bac74e88494
Step 15 : RUN cd /opt/npmjs; couchdb -b; sleep 1; npm run load; sleep 1; curl -k "http://localhost:5984/registry/_design/scratch" -X COPY -H destination:'_design/app'; sleep 1; couchdb -d
 ---> Running in 05206c0ff1c8
Apache CouchDB has started, time to relax.
```

**Comment by terinjokes** [comment]
Yep, could not connect to host when creating the DB. Strange, because it works the next line down.

On the train, but will look as soon as I get in.

**Comment by tj** [comment]
thanks man! I'm going to try using my vm's image and see how that goes

**Comment by tj** [comment]
definitely seems like those sleeps are the racey thing, running them manually works fine

**Comment by terinjokes** [comment]
Yea, that's what I suspected, on EC2 CouchDB doesn't spin up fast enough. Going to try to work around it.

Sorry about the issues, but thanks for finding them

**Comment by tj** [comment]
no worries! should be a great solution for everyone

**Comment by terinjokes** [comment]
FWIW: I just tried to reproduce on the Ubuntu 12.04.3LTS AMI in Northern California setup per http://docs.docker.io/en/latest/installation/ubuntulinux/ and it completed successfully.

**Comment by tj** [comment]
works pretty consistently for me me when I bump the sleeps to 10s

**Comment by terinjokes** [comment]
Works here even with `sleep 1`. Working on waiting on the $PID, but CouchDB isn't being friendly.

**Comment by terinjokes** [comment]
`couchdb -b` writes the PID to /usr/local/var/run/couchdb/couchdb.pid, but this is seconds after the command finishes, which means I would be waiting anyways.

`couchdb &` backgrounds immediately, and you can use the PID at $!. After some amount of time the command name changes from "couchdb" to "beam". I tried detecting this, but found that even after it switched to "beam" the server still wasn't ready to accept HTTP connections.

The CouchDB IRC channel suggests curling / until it's non-error, but that's a lot of extra characters per line, and it's already a bit complicated.

Decided, until there's a better alternative, to simply increase the sleep time. I don't think it will cause that many problems, as I'm going to try to turn on Docker's trusted builds.
